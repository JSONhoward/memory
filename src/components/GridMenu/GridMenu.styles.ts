import * as React from 'react';
import styled from 'styled-components'

export const StyledGridMenu = styled('div')`
display: flex;
height: 50px;
width: 100vh;
margin: 1rem;

@media screen and (orientation: portrait) {
width: 95vw;
}
`

export const DifficultyBox = styled('div')`
display: flex;
align-items: center;
justify-content: space-between;
flex: 2;
font-size: 2vw;
border-radius: 10px;

@media screen and (orientation: portrait) {
    font-size: 5vw;
}
`

export const DifficultyNumbers = React.memo(styled('button')<{disable: boolean}>`
display: grid;
place-items: center;
height: 100%;
flex: 1;
min-width: 25%;
font-size: 1.5vw;
padding: 1px;
margin: .5rem;
border-radius: 5px;
text-transform: uppercase;
box-shadow: 1px 1px 5px rgba(0,0,0,.25);
color: ${props => props.disable ? 'rgba(0,0,0,.25)' : '#373737'};
font-weight: bold;
background-image: linear-gradient(to top, #cfd9df 0%, #e2ebf0 100%);
border: none;
cursor: pointer;

@media screen and (max-width: 400px) {
    font-size: 3vw;
}
`)

export const NewGameBox = React.memo(styled('button')`
display: grid;
place-items: center;
flex: 1;
border-radius: 5px;
font-size: 2vw;
color: #373737;
font-weight: bold;
background-image: linear-gradient(to top, #cfd9df 0%, #e2ebf0 100%);
border: none;
box-shadow: 1px 1px 5px rgba(0,0,0,.25);
margin: 0 .5rem;
cursor: pointer;

@media screen and (max-width: 400px) {
    font-size: 5vw;
}
`)