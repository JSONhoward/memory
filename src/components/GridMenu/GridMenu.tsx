import React from 'react'
import { DifficultyBox, DifficultyNumbers, NewGameBox, StyledGridMenu } from './GridMenu.styles'

type Props = {
    difficulty: string,
    changeDifficulty: (num: string) => void,
    newGame: () => void
}

const GridMenu: React.FC<Props> = ({ difficulty, changeDifficulty, newGame }) => {
    return (
        <StyledGridMenu>
            <DifficultyBox>
                <DifficultyNumbers onClick={() => changeDifficulty('sm')} disable={difficulty !== 'sm'}>easy</DifficultyNumbers>
                <DifficultyNumbers onClick={() => changeDifficulty('md')} disable={difficulty !== 'md'}>normal</DifficultyNumbers>
                <DifficultyNumbers onClick={() => changeDifficulty('lg')} disable={difficulty !== 'lg'}>hard</DifficultyNumbers>
            </DifficultyBox>
            <NewGameBox onClick={newGame}>
                <p>New Game</p>
            </NewGameBox>
        </StyledGridMenu>
    )
}

export default GridMenu
