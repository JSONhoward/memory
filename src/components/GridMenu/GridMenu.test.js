import React from 'react'
import {create} from 'react-test-renderer'
import GridMenu from './'

const props = {
    difficulty: 'sm',
    changeDifficulty: jest.fn(),
    newGame: jest.fn()
}

describe('GridMenu Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<GridMenu />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const component = create(<GridMenu {...props} />).root

        expect(component.props).toEqual(props)
    })
})