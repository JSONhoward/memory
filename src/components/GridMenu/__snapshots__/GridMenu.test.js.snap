// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridMenu Tests Snapshot Test 1`] = `
<div
  className="sc-AxjAm izngQd"
>
  <div
    className="sc-AxirZ kJELKF"
  >
    <button
      className="sc-AxiKw eFtGgs"
      onClick={[Function]}
    >
      easy
    </button>
    <button
      className="sc-AxiKw eFtGgs"
      onClick={[Function]}
    >
      normal
    </button>
    <button
      className="sc-AxiKw eFtGgs"
      onClick={[Function]}
    >
      hard
    </button>
  </div>
  <button
    className="sc-AxhCb gyFxJi"
  >
    <p>
      New Game
    </p>
  </button>
</div>
`;
